k8s/k8s.go

```diff
@@ -0,0 +1,280 @@
+package k8s
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/vuln-list-update/osv"
+	"github.com/aquasecurity/vuln-list-update/utils"
+	uu "github.com/aquasecurity/vuln-list-update/utils"
+)
+
+const (
+	k8svulnDBURL        = "https://kubernetes.io/docs/reference/issues-security/official-cve-feed/index.json"
+	vulnListRepoTarBall = "https://api.github.com/repos/aquasecurity/vuln-list-k8s/tarball"
+	mitreURL            = "https://cveawg.mitre.org/api/cve"
+	cveList             = "https://www.cve.org/"
+)
+
+type VulnDB struct {
+	Cves []*osv.OSV
+}
+
+type CVE struct {
+	Items []Item `json:"items,omitempty"`
+}
+
+type Item struct {
+	ID            string `json:"id,omitempty"`
+	Summary       string `json:"summary,omitempty"`
+	ContentText   string `json:"content_text,omitempty"`
+	DatePublished string `json:"date_published,omitempty"`
+	ExternalURL   string `json:"external_url,omitempty"`
+	URL           string `json:"url,omitempty"`
+}
+
+func Collect() (*VulnDB, error) {
+	response, err := http.Get(k8svulnDBURL)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	var db CVE
+	if err = json.NewDecoder(response.Body).Decode(&db); err != nil {
+		return nil, err
+	}
+	cvesMap, err := getExitingCvesToModifiedMap()
+	if err != nil {
+		return nil, err
+	}
+	return ParseVulnDBData(db, cvesMap)
+}
+
+const (
+	// excludeNonCoreComponentsCves exclude  cves with missing data or non k8s core components
```

Comment:
can you please elaborate on this?

Reply:
this cves which are on non core components, we do not find them with kbom

Reply:
if we don't find them, why do we care if they're in the db?

Reply:
its just matter of keeping the db relevant to our needs and purpose. not more then that